dangerfile.js

```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars
+import { danger, warn, fail, message } from 'danger';
+
+const modifiedChangelog = danger.git.modified_files.filter((filePath) => {
+  const srcFilePattern = /^packages\/([a-z-])*\/CHANGELOG.md/i;
+  return srcFilePattern.test(filePath);
+});
+
+const modifiedSrcFiles = danger.git.modified_files.filter((filePath) => {
+  const srcFilePattern = /^packages\/([a-z-])*\/src/i;
+  return srcFilePattern.test(filePath);
+});
+
+const hasCHANGELOGChanges = modifiedChangelog.length > 0;
+const hasModifiedSrcFiles = modifiedSrcFiles.length > 0;
+
+// Fail if there are src code changes without a CHANGELOG
+if (hasModifiedSrcFiles && !hasCHANGELOGChanges) {
```

Comment:
`danger.git.modified_files` returns an array of strings of the file path + file name of files that have been modified.

The below code is used to filter that array down to only modified files with in the `src` directory of a package. 

The thinking behind this is that there will be trivial changes to docs/tests/etc that likely don't need changelog entries. For those, I'd recommend people use their best judgement. For changes to files in the src directory though, we should be documenting those.

```
const modifiedSrcFiles = danger.git.modified_files.filter((filePath) => {
  const srcFilePattern = /^packages\/([a-z-])*\/src/i;
  return srcFilePattern.test(filePath);
}
```